### DISCUSSION TITLE
Discord message components

### DISCUSSION THREAD

**Discussion by lhy-hoyin** [Rapptz/discord.py#10013] [Q&A] [answered]
I am using slash messages and interactions and I would like to reply with a message which uses [message components](https://discord.com/developers/docs/interactions/message-components). For example:
![image](https://github.com/user-attachments/assets/9c9e2a57-c055-49b9-b6cb-f68462f34950)

However, I am unsure of how the code should look like and would appreciate any hints towards it.

I suppose I can set it up in some kind of view, and then `await interaction.response.send_message(...)` (or maybe `send_modal` or something else) to send it out.

---

I see that https://github.com/Rapptz/discord.py/blob/7db879b5bd7008a28613cf107f434a18b8ea7303/discord/enums.py#L633-L642 matches the [components types](https://discord.com/developers/docs/interactions/message-components) for Discord Dev Docs, so I assume that discord.py supports it.

**Comment by Rapptz** [comment, marked as answer]
Check the [examples](https://github.com/Rapptz/discord.py/tree/master/examples/views) on how to send views.
